HACKING.md

```diff
@@ -45,21 +45,68 @@ source directory toplevel.  You can provision a VM however you want; libvirt
 directly, vagrant, a remote OpenStack/EC2 instance, etc.  If you choose
 vagrant for example, do something like this:
 
-```
+```sh
 vagrant ssh-config > /path/to/src/rpm-ostree/ssh-config
 ```
 
+The host is expected to be called `vmcheck` in the
+`ssh-config`. You can specify multiple hosts and parallelize
+the `make vmcheck` testsuite run through the `HOSTS`
+variable. For example, if you have three nodes named
+`vmcheck[123]`, you can use:
+
+```sh
+make vmcheck HOSTS='vmcheck1 vmcheck2 vmcheck3'
+```
+
 Once you have a `ssh-config` set up:
 
 `make vmsync` will do an unlock, and sync the container build
 into the VM.
 
 `make vmoverlay` will do a non-live overlay, and reboot the VM.
 
-For convenience, the `make vmshell` command does the same
-as `make vmsync` but additionally places you in a shell,
-ready to test your changes.
+Note that by default, these commands will retrieve the latest version of ostree
+from the build environment and include those binaries when syncing to the VM.
+
+Ideally, you should be installing `ostree` from streams like
+[FAHC](https://pagure.io/fedora-atomic-host-continuous/) and
+[CAHC](https://wiki.centos.org/SpecialInterestGroup/Atomic/Devel), which closely
+track ostree's git master. This allows you to not have to worry about using
+libostree APIs that are not yet released.
+
+Testing with a custom libdnf
+============================
+
+Since rpm-ostree uses libdnf as a submodule for the time being, you can simply
+edit the libdnf submodule directly and the various `make` targets will pick up
+the changes and recompile.
+
+Testing with a custom ostree
+============================
+
+It is sometimes necessary to develop against a version of ostree which is not
+even yet in git master. In such situations, one can simply do:
```

Comment:
Seems OK to me, though I personally just do `sudo make install` from ostree git, but that's just into my dev container.

Reply:
Yeah, though the issue I had with that is that I had to do `touch insttree/.completed` which just felt hacky and completely unnecessary. It also assumes that the file list didn't change since the last FAHC/CAHC build.